src/frontend/src/components/Post/Post.jsx

```diff
@@ -81,9 +92,27 @@ const Post = ({ id, html, author, url, title, date, link }) => {
     syntaxHighlight(sectionEl.current);
   }, [sectionEl]);
 
+  function displayPostButtons() {
+    return (
+      <div className={classes.iconDiv}>
+        <span className={classes.iconSpan}>
+          <IconButton size="small">
+            <FlagIcon fontSize="large" />
+          </IconButton>
+        </span>
+        <span className={classes.iconSpan}>
+          <IconButton size="small">
+            <DeleteIcon fontSize="large" />
+          </IconButton>
+        </span>
+      </div>
+    );
+  }
+
   return (
     <Container className={classes.root}>
       <ListSubheader className={classes.header}>
+        {user.isAdmin ? displayPostButtons() : null}
```

Comment:
This should be done with a new component, and let it decide to render or not based on the `user` prop:

```jsx
<AdminButtons user={user} />
```

Then create a new component `admin-buttons.jsx` and refactor your function above into a function component:

```jsx
function AdminButtons({ user }) {
    if(!user.isAdmin) {
      return null;
    }

    return (
      <div className={classes.iconDiv}>
        <span className={classes.iconSpan}>
          <IconButton size="small">
            <FlagIcon fontSize="large" />
          </IconButton>
        </span>
        <span className={classes.iconSpan}>
          <IconButton size="small">
            <DeleteIcon fontSize="large" />
          </IconButton>
        </span>
      </div>
    );
}
```

Reply:
Got it, I opted to not pass user as a prop since we have the `UserStateContext` that can be used